src/CodeViewer/Viewer.cpp

```diff
@@ -28,27 +29,37 @@ static const QColor kTextEditBackgroundColor{30, 30, 30};
 static const QColor kTextEditForegroundColor{189, 189, 189};
 static const QColor kHeatmapColor{Qt::red};
 
+int StringWidthInPixels(const QFontMetrics& font_metrics, std::string string) {
+  return font_metrics.horizontalAdvance(QString::fromStdString(string));
+}
+
 int DetermineLineNumberWidthInPixels(const QFontMetrics& font_metrics, int max_line_number) {
-  return font_metrics.horizontalAdvance(QString::number(max_line_number));
+  return StringWidthInPixels(font_metrics, std::to_string(max_line_number));
 }
 
-Viewer::Viewer(QWidget* parent) : QPlainTextEdit(parent), left_sidebar_widget_{this} {
-  UpdateLeftSidebarWidth();
-  QObject::connect(this, &QPlainTextEdit::blockCountChanged, this, &Viewer::UpdateLeftSidebarWidth);
+Viewer::Viewer(QWidget* parent)
+    : QPlainTextEdit(parent), left_sidebar_widget_{this}, right_sidebar_widget_{this} {
+  UpdateSidebarsWidth();
+  QObject::connect(this, &QPlainTextEdit::blockCountChanged, this, &Viewer::UpdateSidebarsWidth);
 
   QObject::connect(&left_sidebar_widget_, &PlaceHolderWidget::PaintEventTriggered, this,
                    &Viewer::DrawLineNumbers);
 
+  QObject::connect(&right_sidebar_widget_, &PlaceHolderWidget::PaintEventTriggered, this,
+                   &Viewer::DrawSampleCounters);
+
   const auto update_viewport_area = [&](const QRect& rect, int dy) {
     const auto update_caused_by_scroll = [](int dy) { return dy != 0; };
```

Comment:
I'm not very clear about the purpose for using a lambda expression here. `update_cuased_by_scroll` seems only used in line 54. Why we are not using a boolean `update_cuased_by_scroll` here?

Reply:
This was here before, and I also found it strange. I think that it was used to give the user a better explanation than a simply if(dy != 0). I changed it, What do you think?

Reply:
Thanks for the explanation and making the change :-) I understand the purpose for the readability. But just curious why not a simple `bool update_cuased_by_scroll = (dy != 0)` like we are using in other cases.

Reply:
Nice suggestion!! I don't know why, but we are using it now :)